File: app/src/main/java/com/spdroid/schedulefcm/example/fcm/NotificationBroadcastReceiver.java
package com.spdroid.schedulefcm.example.fcm;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.work.Data;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkManager;

import static com.spdroid.schedulefcm.example.fcm.ScheduledWorker.NOTIFICATION_MESSAGE;
import static com.spdroid.schedulefcm.example.fcm.ScheduledWorker.NOTIFICATION_TITLE;


/**
 * Created by rogergcc on 22/02/2020.
 * Copyright Ⓒ 2020 . All rights reserved.
 */
public class NotificationBroadcastReceiver extends BroadcastReceiver {
    public void onReceive(@Nullable Context context, @Nullable Intent intent) {
        if (intent != null) {
            if (intent.getExtras() == null) {
                return;
            }
            String title = intent.getStringExtra(NOTIFICATION_TITLE);
            String message = intent.getStringExtra(NOTIFICATION_MESSAGE);

            // Create Notification Data
            Data notificationData  = (new Data.Builder()).putString(NOTIFICATION_TITLE, title).putString(NOTIFICATION_MESSAGE, message).build();
            if (notificationData  == Data.EMPTY) {
                return;
            }
            // Init Worker
            OneTimeWorkRequest work = ((new OneTimeWorkRequest.Builder(ScheduledWorker.class)).setInputData(notificationData )).build();

            // Start Worker
            WorkManager.getInstance().beginWith(work).enqueue();
            Log.d(this.getClass().getName(), "WorkManager is Enqueued.");
        }
    }
}

File: app/src/main/java/com/spdroid/schedulefcm/example/fcm/NotificationBroadcastReceiver.kt
package com.spdroid.schedulefcm.example.fcm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import androidx.work.Data
import androidx.work.OneTimeWorkRequest
import androidx.work.WorkManager
import com.spdroid.schedulefcm.example.fcm.NotificationScheduler.Companion.NOTIFICATION_MESSAGE
import com.spdroid.schedulefcm.example.fcm.NotificationScheduler.Companion.NOTIFICATION_TITLE

class NotificationBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context?, intent: Intent?) {
        intent?.let {
            val title = it.getStringExtra(NOTIFICATION_TITLE)
            val message = it.getStringExtra(NOTIFICATION_MESSAGE)

            // Create Notification Data
            val notificationData = Data.Builder()
                .putString(NOTIFICATION_TITLE, title)
                .putString(NOTIFICATION_MESSAGE, message)
                .build()

            // Init Worker
            val work = OneTimeWorkRequest.Builder(NotificationScheduler::class.java)
                .setInputData(notificationData)
                .build()

            // Start Worker
            WorkManager.getInstance().beginWith(work).enqueue()

            Log.d(javaClass.name, "WorkManager is Enqueued.")
        }
    }
}
